### PR TITLE
Ask for typesafe id in a popup before cloning

### PR DESCRIPTION
**Pull request by Warry**
if the chosen template is TRP, then it asks for an ID then send it with the `CreateNewApplication` message (the associate json object has a `typesafeId` field).

### CODE REVIEW

File: ui/app/assets/widgets/home/templates/templates.js [hunk 1]
```diff
@@ -6,19 +6,50 @@ define([
...
+    if (e.target.value.length === 36){
+      activeOk(true);
+      typesafeId(e.target.value);
+    } else {
+      activeOk(false);
+      typesafeId("");
+    }
+  }
+  function askForTypesafeId(callback){
+    var message = $("<article/>").html("<p>You are creating Typesafe Reactive Platform project, which require a Typesafe ID.</p><p>You can retreive you ID on the <a href='http://typesafe.com/subscription' target='_blanc'>typesafe.com/subscription</a> page.<p><p class='input'></p>")[0];
```
Comment (havocp) [inline ui/app/assets/widgets/home/templates/templates.js]: s/require/requires/ s/retreive/retrieve/ s/you/your/

We should suggest the same page as the RP plugin's error in the logs; does it suggest /subscription or /account/id ?

Comment (havocp) [inline ui/app/assets/widgets/home/templates/templates.js]: Let's say "retrieve your ID or sign up for a free trial" ?

File: ui/app/assets/widgets/home/templates/templates.js [hunk 2]
```diff
@@ -6,19 +6,50 @@ define([
...
+  tpl,
+  modals
 ) {
 
   var trpInfoSeen = settings.observable("reactive-platform.accepted-license", false);
 
+  var typesafeId = settings.observable("TypesafeID", "");
+  var activeOk = ko.observable(!!typesafeId());
+  function saveTypesafeID(e){
+    if (e.target.value.length === 36){
```
Comment (havocp) [inline ui/app/assets/widgets/home/templates/templates.js]: trimming whitespace off the string for people might save some confusion for them

**Comment by havocp** [comment]
I have the card to implement the backend for this, which should be simple enough - will do as soon as this is merged.

File: ui/app/assets/widgets/home/templates/templates.js [hunk 3]
```diff
@@ -6,19 +6,50 @@ define([
...
 ], function(
   settings,
   websocket,
   FileSelection,
-  tpl
+  tpl,
+  modals
 ) {
 
   var trpInfoSeen = settings.observable("reactive-platform.accepted-license", false);
```
Comment (havocp) [inline ui/app/assets/widgets/home/templates/templates.js]: btw is this now obsolete? I guess our current idea is that if you have a subscription ID, that means you accepted the license (you have to accept in order to get an ID)

File: ui/app/assets/widgets/home/templates/templates.js [hunk 4]
```diff
@@ -6,19 +6,50 @@ define([
...
 
+  var typesafeId = settings.observable("TypesafeID", "");
+  var activeOk = ko.observable(!!typesafeId());
+  function saveTypesafeID(e){
+    if (e.target.value.trim().length === 36){
+      activeOk(true);
+      typesafeId(e.target.value);
```
Comment (havocp) [inline ui/app/assets/widgets/home/templates/templates.js]: you trim before validating but then haven't trimmed what you store here, right?

File: ui/app/assets/widgets/home/templates/templates.js (hunk 3 above)
Comment (Warry) [inline ui/app/assets/widgets/home/templates/templates.js]: We still show the pop up on the very first visit of that tab !

Comment (havocp) [inline ui/app/assets/widgets/home/templates/templates.js]: ah, ok. let's leave it alone for now, not mix with this PR

**Pull request merged**